### DISCUSSION TITLE
"Import scores from stable" not showing ranks

### DISCUSSION THREAD

**Discussion by Valsitsar** [ppy/osu#16259] [Q&A]
So I downloaded osu!lazer earlier today, and started importing things from the stable version. Beatmaps imported more-or-less without issues, other than the fact that some of them generated some kind of error message, but there were few enough of those that I'd decided to leave that be. However, when I tried to import the scores from stable, it said the game was importing some 2k+ scores, which seemed about an accurate number, given that I had imported some 300 beatmaps or so. The issue is that, once the process had finished, my achieved ranks on the individual beatmaps were nowhere to be seen (I mean when scrolling through the list of beatmaps). On top of that, once I finish a beatmap anew, the rank I get on that attempt gets shown on the list, even though on stable, my top score on that map is an S, for example. I wonder if I'm still missing something, or if this is intended, or what's up. Any advice is appreciated, and I thank you in advance for replying.

**Comment by integerrr** [comment]
Lazer currently doesnt show your achieved rank on the beatmap carousel like stable does.
I dont think I understand your second question

**Reply by bdach** [reply]
> Lazer currently doesnt show your achieved rank on the beatmap carousel like stable does.

uhh...

![2021-12-27-091507_469x403_scrot](https://user-images.githubusercontent.com/20418176/147452127-86ed7ce6-1767-4c4d-8cbd-ee9c4b143fd0.png)

the *actual* problem is that stable replays don't have the user's online ID inside them, just the username. and the local user rank display uses the online ID to determine if a score is yours. the fix would be to map usernames from stable scores back to online IDs, *if possible* (which is not a given, because renames are a thing)
